logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java

```diff
@@ -45,7 +48,9 @@ public interface PipelineBus {
     static PipelineBus create() {
         final String pipelineBusImplementation = System.getProperty("logstash.pipelinebus.implementation", "v2");
         switch (pipelineBusImplementation) {
-            case "v1": return new PipelineBusV1();
+            case "v1":
+                DEPRECATION_LOGGER.deprecated("The legacy pipeline bus selected with `logstash.pipelinebus.implementation=v1` is deprecated, and will be removed in a future release of Logstash");
```

Comment:
It seems like configuring pipeline bus implementation setting itself is or should be deprecated. Maybe the deprecation should be focused on removing that setting rather than the value it is set to.